sympy/solvers/tests/test_solveset.py

```diff
@@ -3077,3 +3077,12 @@ def test_issue_22058():
 
 def test_issue_11184():
     assert solveset(20*sqrt(y**2 + (sqrt(-(y - 10)*(y + 10)) + 10)**2) - 60, y, S.Reals) is S.EmptySet
+
+
+def test_issue_21890():
+    x, y = symbols('x y', real=True)
+    sol = nonlinsolve([4*x**3*y**4 - 2*y, 4*x**4*y**3 - 2*x], x, y)
```

Comment:
Are all three error cases working? That is, this but without the real assumption, and the one in a later comment?

Reply:
checking now...

Reply:
yes, all three cases work. Tests have been added.